packages/api/src/core/prepareAuth.ts

```diff
@@ -18,25 +16,63 @@ export default function prepareAuth(authKey: (number | string)[], operation: Ope
       }
   > = {};
 
-  const security = operation.prepareSecurity();
-
-  const securitySchemes = Object.keys(security);
-  if (securitySchemes.length === 0) {
+  const security = operation.getSecurity();
+  if (security.length === 0) {
     // If there's no auth configured on this operation, don't prepare anything (even if it was
     // supplied by the user).
     return {};
   }
 
-  const securityType = securitySchemes[0] as SecurityType;
-
-  const schemes = security[securityType];
+  // Does this operation require multiple forms of auth?
+  if (security.every(s => Object.keys(s).length > 1)) {
+    throw new Error(
+      "Sorry, this operation currently requires multiple forms of authentication which this library doesn't yet support."
+    );
+  }
 
-  if (schemes.length > 1) {
-    throw new Error("Sorry, this API currently requires multiple forms of authentication which we don't yet support.");
+  // Since we can only handle single auth security configurations, let's pull those out. This code
+  // is a bit opaque but `security` here may look like `[{ basic: [] }, { oauth2: [], basic: []}]`
+  // and are filtering it down to only single-auth requirements of `[{ basic: [] }]`.
+  const usableSecurity = security
+    .map(s => {
+      return Object.keys(s).length === 1 ? s : false;
+    })
+    .filter(Boolean);
+
+  const usableSecuritySchemes = usableSecurity.map(s => Object.keys(s)).reduce((prev, next) => prev.concat(next), []);
+  const preparedSecurity = operation.prepareSecurity();
+
+  // If we have two auth tokens present let's look for Basic Auth in their configuration.
+  if (authKey.length >= 2) {
+    // If this operation doesn't support HTTP Basic auth but we have two tokens, that's a paddlin.
+    if (!('Basic' in preparedSecurity)) {
+      throw new Error('Multiple auth tokens were supplied for this endpoint but only a single token is needed.');
+    }
+
+    const schemes = preparedSecurity.Basic.filter(s => usableSecuritySchemes.includes(s._key));
+    if (!schemes.length) {
```

Comment:
I'm not entirely sure what you're trying to check for here but I don't think this matches with the error message on the next line.

Reply:
Yeah good catch, I've revised the error message to be clearer and added a comment here explaining what the hell I'm doing.